monitor/statistics.hpp

```diff
@@ -28,6 +28,10 @@
 #include <mutex>
 #include <sstream>
 #include <thread>
+#include <map>
+
+#include <boost/thread/locks.hpp>
+#include <boost/thread/shared_mutex.hpp>
```

Comment:
shared_mutex - is a wrapper around pthread_rwlock_t, so statistics::report could be accessed without exclusive locks. Also std::shared_mutex has appeared in C++14.

Ok, I will remove this. Will be slower, but without boost.